src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/entity/AssessmentStatus.kt

```diff
@@ -187,8 +221,19 @@ enum class AssessmentStatus {
   ;
 
   open fun transitions(): Set<Transition> = emptySet()
+
+  open fun tasks(): Set<TaskProgress> = emptySet()
 }
 
 fun transitions(vararg states: AssessmentStatus) = states.map { Transition(to = it) }.toSet()
 
 data class Transition(val to: AssessmentStatus)
+
+sealed interface TaskProgress {
+  val task: Task
+  val status: (assessment: Assessment) -> TaskStatus
```

Comment:
little bit of support to allow dynamically calculating status of task based on additional data though nothing is using this currently